tasks/clientlibs.js

```diff
@@ -377,17 +387,95 @@ module.exports = function (grunt) {
 		 * @returns {object} The sorted array
 		 */
 		function performSort(arr) {
-			var returnValue = null,
-				arrLength = arr.length,
-				i = 0;
+			var arrLength = arr.length,
+				i = 0,
+				resultArray = [],
+				item = {};				
+				
+			for (i = 0; i < arrLength; i++) {
+				item = arr[i];
+				if (checkIfAlreadySortedForFile(item, resultArray) === false) {
+					sortSpecifiFile(item, arr, resultArray);
+				}				
+			}
+			
+			return resultArray;
+		}
+		/**
+		 * @function getDependFile
+		 * @description A helper method to check if have depends in this client libs
+		 * param {string} dependFileName
+		 * param {array} remainArray
+		 * returns {object} return null if there is no dependencies in this library, file return specific dependency file 
+		 */
+		function getDependFile(dependFileName, remainArray) {
+			var i = 0,
+				len = remainArray.length,
+				dependName = '',
+				specificFileName = '';				
+
+			for (i = 0; i < len; i++) {								
```

Comment:
Same comment as above. Nothing make or break, but if we are keeping to a standard, this would be one.

Reply:
Updated